src/angular-public/form-field/form-field.ts

```diff
@@ -0,0 +1,213 @@
+  selector: 'sbb-form-field, sbb-field',
+  exportAs: 'sbbFormField',
+  templateUrl: './form-field.html',
+  styleUrls: ['./form-field.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  providers: [{ provide: SBB_FORM_FIELD, useExisting: SbbFormField }],
+  host: {
+    class: 'sbb-form-field',
+    '[class.sbb-form-field-default]': 'this.mode === "default"',
+    '[class.sbb-form-field-short]': 'this.mode === "short"',
+    '[class.sbb-form-field-medium]': 'this.mode === "medium"',
+    '[class.sbb-form-field-long]': 'this.mode === "long"',
+    '[class.sbb-form-field-invalid]': '_control?.errorState',
+    '[class.sbb-form-field-disabled]': '_control?.disabled',
+    '[class.sbb-focused]': '_control?.focused',
+    '[class.ng-untouched]': '_shouldForward("untouched")',
+    '[class.ng-touched]': '_shouldForward("touched")',
+    '[class.ng-pristine]': '_shouldForward("pristine")',
+    '[class.ng-dirty]': '_shouldForward("dirty")',
+    '[class.ng-valid]': '_shouldForward("valid")',
+    '[class.ng-invalid]': '_shouldForward("invalid")',
+    '[class.ng-pending]': '_shouldForward("pending")',
+  },
+})
+export class SbbFormField implements AfterContentInit, AfterContentChecked, OnDestroy {
+  /** The label text for the input. */
+  @Input() label?: string;
+  /**
+   * Mode set the length of the input field.
+   * @deprecated Use corresponding css classes instead (.sbb-form-field-short, .sbb-form-field-medium, .sbb-form-field-long).
+   * @breaking-change 12.0.0
+   */
+  @Input()
+  get mode(): 'default' | 'short' | 'medium' | 'long' {
+    return this._mode;
+  }
+  set mode(value: 'default' | 'short' | 'medium' | 'long') {
+    if (typeof ngDevMode === 'undefined' || ngDevMode) {
+      console.warn(
+        `${this._elementRef.nativeElement.nodeName.toLowerCase()}[mode] is deprecated! ` +
+          'Use the corresponding css classes instead ' +
+          '(.sbb-form-field-short, .sbb-form-field-medium, .sbb-form-field-long).'
+      );
+    }
+    this._mode = value;
+  }
+  private _mode: 'default' | 'short' | 'medium' | 'long' = 'default';
+
+  private _destroyed = new Subject<void>();
+
+  // Unique id for the label element.
+  readonly _labelId = `sbb-form-field-label-${nextId++}`;
+
+  @ContentChild(SbbFormFieldControl) _controlNonStatic: SbbFormFieldControl<any>;
+  @ContentChild(SbbFormFieldControl, { static: true }) _controlStatic: SbbFormFieldControl<any>;
+  get _control() {
+    // TODO(crisbeto): we need this workaround in order to support both Ivy and ViewEngine.
+    //  We should clean this up once Ivy is the default renderer.
+    return this._explicitFormFieldControl || this._controlNonStatic || this._controlStatic;
+  }
+  set _control(value) {
+    this._explicitFormFieldControl = value;
+  }
+  private _explicitFormFieldControl: SbbFormFieldControl<any>;
+
+  @ContentChild(SbbLabel) _labelChildNonStatic: SbbLabel;
```

Comment:
can be private, not? (and following 2 lines)

Reply:
I don't think that this would be compatible for ViewEngine.